app/components/Upload/FileUpload.tsx

```diff
@@ -1,72 +1,56 @@
+import { Button, Icon } from '@webkom/lego-bricks';
+import { UploadIcon } from 'lucide-react';
+import { useState, useRef } from 'react';
 import { connect } from 'react-redux';
 import { uploadFile } from 'app/actions/FileActions';
 import styles from './FileUpload.css';
 
-type State = {
-  pending: boolean;
-};
-type Props = {
-  uploadFile: (arg0: { file: File }) => Promise<any>;
+type FileUploadProps = {
+  uploadFile: (arg0: { file: File }) => Promise<{
+    meta: { fileKey: string; fileToken: string };
+  }>;
   onChange: (arg0: string, arg1: string) => void;
   className?: string;
 };
 
-class FileUpload extends Component<Props, State> {
-  input: HTMLInputElement | null | undefined;
-  state = {
-    pending: false,
-  };
-  handleClick = () => {
-    this.input && this.input.click();
+const FileUpload = ({ uploadFile, onChange, className }: FileUploadProps) => {
+  const [pending, setPending] = useState(false);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleClick = () => {
+    inputRef.current && inputRef.current.click();
   };
-  handleChange = (e) => {
-    const file = e.target.files[0];
-    this.setState({
-      pending: true,
-    });
-    this.props
-      .uploadFile({
-        file,
-      })
-      .then(({ meta }) => {
-        this.setState({
-          pending: false,
-        });
-        this.props.onChange(meta.fileKey, meta.fileToken);
-      })
-      .catch((error) => {
-        this.setState({
-          pending: false,
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0];
+    if (file) {
+      setPending(true);
+      uploadFile({ file })
+        .then(({ meta }) => {
+          setPending(false);
+          onChange(meta.fileKey, meta.fileToken);
+        })
+        .catch((error) => {
+          setPending(false);
+          throw error;
         });
-        throw error;
-      });
+    }
   };
 
-  render() {
-    return (
-      <div>
-        <Button
-          isPending={this.state.pending}
-          onPress={this.handleClick}
-          className={this.props.className}
-        >
-          Last opp fil
-        </Button>
-        <input
-          ref={(ref) => (this.input = ref)}
-          className={styles.input}
-          onChange={this.handleChange}
-          type="file"
-          accept="application/pdf"
-        />
-      </div>
-    );
-  }
-}
-
-const mapDispatchToProps = {
-  uploadFile,
+  return (
+    <div>
+      <Button isPending={pending} onPress={handleClick} className={className}>
+        <Icon name="upload" iconNode={<UploadIcon size={24} />} />
+      </Button>
+      <input
+        ref={inputRef}
+        className={styles.input}
+        onChange={handleChange}
+        type="file"
+        accept="application/pdf"
+      />
+    </div>
+  );
 };
-export default connect(null, mapDispatchToProps)(FileUpload);
+
+export default connect(null, { uploadFile })(FileUpload);
```

Comment:
Remove the `connect` from here and call `uploadFile` using `useAppDispatch` inside the component.

```typescript
const FileUpload = ({ onChange, className }: FileUploadProps) => {
	dispatch = useAppDispatch();
	...
 		dispatch(uploadFile)
	...
```